### PR TITLE
Add complete interpreter test coverage for Policy

### PR DESCRIPTION
**Pull request by Abhay349**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
This PR adds comprehensive test coverage for the Policy resource interpreter customization.

Specifically, it:
	•	Adds unit tests for AggregateStatus, InterpretStatus, and InterpretHealth
	•	Aligns test expectations with the actual aggregation behavior 
	
These tests improve confidence in Kyverno Policy status handling across clusters and prevent future regressions.

**Which issue(s) this PR fixes**:
Fixes part of #6952

**Special notes for your reviewer**:
	•	Tests are written to strictly follow the existing Lua aggregation logic
	•	All cases are verified using:
                 `go test ./pkg/resourceinterpreter/default/thirdparty`

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

**Comment by FAUST-BENCHOU** [comment]
/assign 
may takes some time

**Comment by XiShanYongYe-Chang** [comment]
> Fixes https://github.com/karmada-io/karmada/issues/6952

Hi @Abhay349, this info can be updated to:

```
Fixes part of #xxx
```

As it is currently written, when a PR is merged, the associated issue will also be closed by GitHub. This is appropriate for individual issues, but for umbrella issues, a single PR often resolves only a subset of the problems in most cases.

**Comment by FAUST-BENCHOU** [comment]
ya we still get one pr to track

**Comment by Ferryx349** [comment]
> > Fixes #6952
> 
> Hi @Abhay349, this info can be updated to:
> 
> ```
> Fixes part of #xxx
> ```
> 
> As it is currently written, when a PR is merged, the associated issue will also be closed by GitHub. This is appropriate for individual issues, but for umbrella issues, a single PR often resolves only a subset of the problems in most cases.

@XiShanYongYe-Chang Thanks, I will keep this in mind from next times.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/aggregatestatus-test.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # test case for aggregating status of Policy
 # case1. Policy with two status items
+# case2. Policy with different Ready reasons
+# case3. Policy with nil statusItems
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/aggregatestatus-test.yaml:4]: conseder add a mixed test that a certain statusItem has no status or its status is empty.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/interpretstatus-test.yaml [hunk 2]
```diff
@@ -92,3 +95,107 @@ observedObj:
...
+        type: Ready
+    ready: true
+    rulecount:
+      generate: 0
+      mutate: 0
+      validate: 1
+      verifyimages: 0
+---
+name: "Policy with nil status"
+description: "InterprettStatus should return empty status when status is nil"
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/interpretstatus-test.yaml]: spell error

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/interpretstatus-test.yaml [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # test case for interpreting status of Policy
 # case1. Policy: interpret status test
+# case2. Policy with nil status
+# case3. Policy with only ready field
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/interpretstatus-test.yaml:4]: Besides Policy with only ready field we can also add only conditions and only autogen case
https://github.com/karmada-io/karmada/blob/cd8e250d73db8864eee00687cad839d5d8a085a9/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/customizations.yaml#L82-L83

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/interprethealth-test.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
+# testcases for interprethealth of Policy
+# case1. Policy with status.ready = true
+# case2. Policy with status.ready false
+# case3. Policy with Ready=True and Succeeded condition
+# case4. Policy with Ready=True but non-succeeded reason
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/Policy/testdata/interprethealth-test.yaml:5]: nit: The test case here covers "there is a Ready condition but the reason is incorrect"; consider adding a test to cover "there is no Ready condition at all" (status.conditions is not empty, but there is no type=Ready).

**Comment by Ferryx349** [comment]
Hi @FAUST-BENCHOU thanks for review and feedback and apologies for delay, I have fixed the recommended changes, let me know if there are issues..
Happy to iterate on further feedback..
Thanks again :)

**Comment by FAUST-BENCHOU** [comment]
lgtm.Thanks for your work!

**Comment by XiShanYongYe-Chang** [comment]
Thanks @Abhay349 @FAUST-BENCHOU 
/retest
